### DISCUSSION TITLE
Can Turborepo work with Deno?

### DISCUSSION THREAD

**Discussion by grantjayy** [vercel/turborepo#3057] [Help] [answered]
Couldn’t find anything on this topic so asking here. Will turborepo work with Deno? Or is it strictly for the Node runtime?

**Comment by mehulkar** [comment, marked as answer]
Turborepo does not really use the Node runtime, it's a standalone binary (written in Go and Rust). I'm not familiar with Deno, but you're likely to run into issues unless it uses the same `package.json` format and one of our support package managers for Node.

**Comment by akoenig** [comment]
Deno [1.31](https://deno.com/blog/v1.31) now supports `package.json` manifests out of the box. I didn't test it yet, but it _should_ be possible to use `turbo` with `deno` now.

**Comment by lveillard** [comment]
I played with this a bit. it does work when using the package.json, but ideally there could be an option in turborepo to read the scripts from a different file. For instance using deno with aleph, the default "package" manager is deno.json which does not have the format.

```
{
  "compilerOptions": {
    "lib": ["dom", "dom.iterable", "dom.extras", "deno.ns"],
    "types": ["https://deno.land/x/aleph@1.0.0-beta.44/types.d.ts"],
    "jsx": "react-jsx",
    "jsxImportSource": "https://esm.sh/v128/react@18.2.0"
  },
  "importMap": "import_map.json",
  "tasks": {
    "dev": "deno run -A https://deno.land/x/aleph@1.0.0-beta.44/dev.ts",
    "start": "deno run -A server.ts",
    "build": "deno run -A server.ts --build",
    "esm:add": "deno run -A https://esm.sh/v128 add",
    "esm:update": "deno run -A https://esm.sh/v128 update",
    "esm:remove": "deno run -A https://esm.sh/v128 remove"
  }
  ```
  
  Two issues of using the package.json format:
  - Issues with ts compiler that adds errors everywhere 😅
  - Not sure how turbo will be able to get the topological order of the dependencies, as the format is not the same. 
  
  Will keep playing with it in case i make it work, but maybe we will be forced to jump to nx :'(
